### PR TITLE
Clean up tests and add a few new ones

### PR DESCRIPTION
**Pull request by kfichter**
## Description
Adds various tests where clearly missing (as per coverage reports) in preparation for an internal audit. Also cleans up some of the test helpers to clarify a few things.

### Fixes
- Fixes [YAS-538](https://optimists.atlassian.net/browse/YAS-538), [YAS-540](https://optimists.atlassian.net/browse/YAS-540), [YAS-557](https://optimists.atlassian.net/browse/YAS-557)

## Contributing Agreement

- [x] I have read and understood the [Optimism Contributing Guide and Code of Conduct](https://github.com/ethereum-optimism/optimism-monorepo/blob/master/.github/CONTRIBUTING.md) and am following those guidelines in this pull request.

### CODE REVIEW

File: packages/contracts/test/test-helpers/constants.ts [hunk 1]
```diff
@@ -52,3 +27,21 @@ export const L2_TO_L1_MESSAGE_PASSER_OVM_ADDRESS =
 export const CHAIN_ID = 108
 export const ZERO_UINT = '00'.repeat(32)
 export const DEFAULT_FORCE_INCLUSION_PERIOD = 600
+
+export const DEFAULT_UNSAFE_OPCODES = UNSAFE_OPCODES.concat([Opcode.CHAINID])
```
Comment (K-Ho) [inline packages/contracts/test/test-helpers/constants.ts:31]: Why is `CHAINID` concatenated here? Should we just add `CHAINID` to `UNSAFE_OPCODES`?

File: packages/contracts/test/contracts/chain/CanonicalTransactionChain.spec.ts [hunk 2]
```diff
@@ -752,9 +851,54 @@ describe('CanonicalTransactionChain', () => {
       elementInclusionProof.indexInBatch++
       const isIncluded = await canonicalTxChain.verifyElement(
         element,
-        wrongPosition,
+        position,
```
[the remark this answers is not in the document]
Comment (kfichter) [reply packages/contracts/test/contracts/chain/CanonicalTransactionChain.spec.ts:854]: Yep, whoops

File: packages/contracts/test/contracts/chain/CanonicalTransactionChain.spec.ts [hunk 3]
```diff
@@ -752,9 +851,54 @@ describe('CanonicalTransactionChain', () => {
...
+        canonicalTxChain,
+        sequencer,
+        batch
+      )
+      const elementIndex = 1
+      const element = batch[elementIndex]
+      const position = localBatch.getPosition(elementIndex)
+      const elementInclusionProof = await localBatch.getElementInclusionProof(
+        elementIndex + 1 // Proof for the wrong thing
+      )
```
[the remark this answers is not in the document]
Comment (kfichter) [reply packages/contracts/test/contracts/chain/CanonicalTransactionChain.spec.ts:894]: Ah good point here.

File: packages/contracts/test/test-helpers/constants.ts (hunk 1 above)
Comment (kfichter) [reply packages/contracts/test/test-helpers/constants.ts:31]: `UNSAFE_OPCODES` comes from `@eth-optimism/rollup-core`. I didn't know if there was a reason why `CHAINID` wasn't part of `UNSAFE_OPCODES` there, didn't want to touch it.

**Comment by kfichter** [comment]
Closing this because it's not worth the trouble to fix the merge conflicts.

**Pull request closed without merging**